src/loaders/OMETiffLoader.js

```diff
@@ -324,87 +117,72 @@ export default class OMETiffLoader {
     };
   }
 
-  async _getChannel({ image, x, y, z, signal }) {
-    const { tileSize, pool, isInterleaved: interleave } = this;
-    const { height, width } = this._getTileExtent({
-      x,
-      y,
-      z
-    });
-    // Passing in the height and width explicitly prevents resampling that geotiff does without such parameters.
-    const rasters = await image.readRasters({
-      window: [
-        x * tileSize,
-        y * tileSize,
-        x * tileSize + width,
-        y * tileSize + height
-      ],
-      pool,
-      signal,
-      width,
-      height,
-      interleave
-    });
-    const data = interleave ? rasters : rasters[0];
-    if (signal?.aborted) return null;
-    if (this.omexml.Type.startsWith('int')) {
-      // Uint view isn't correct for underling buffer, need to take an
-      // IntXArray view and cast to UintXArray.
-      if (data.BYTES_PER_ELEMENT === 1) {
-        return new Uint8Array(new Int8Array(data.buffer));
+  _getIFDIndex({ z = 0, channel, time = 0 }, zIndex = 0) {
+    let channelIndex;
+    // Without names, enforce a numeric channel indexing scheme
+    const channelNames = this.dimensions.find(dim => dim.field === 'channel')
+      .values;
+    if (channelNames.every(v => !v)) {
+      console.warn(
+        'No channel names found in OMEXML.  Please be sure to use numeric indexing.'
+      );
+      channelIndex = channel;
+    } else if (typeof channel === 'string') {
+      channelIndex = channelNames.indexOf(channel);
+    } else if (typeof channel === 'number') {
+      channelIndex = channel;
+    } else {
+      throw new Error('Channel selection must be numeric index or string');
+    }
+    const sizeZ = this.dimensions.find(dim => dim.field === 'z').values.length;
+    const sizeT = this.dimensions.find(dim => dim.field === 'time').values
+      .length;
+    const sizeC = this.dimensions.find(dim => dim.field === 'channel').values
+      .length;
+    const dimensionOrder = this.dimensions
+      .map(dim => dim.field[0])
+      .reverse()
+      .join('');
+    const pyramidOffset = zIndex * sizeZ * sizeT * sizeC;
+    switch (dimensionOrder) {
+      case 'xyzct': {
+        return pyramidOffset + time * sizeZ * sizeC + channelIndex * sizeZ + z;
       }
-      if (data.BYTES_PER_ELEMENT === 2) {
-        return new Uint16Array(new Int16Array(data.buffer));
```

Comment:
These diffs are _really_ hard to read through. Where did this logic go? It's super hard to tell what got changed or if something didn't get copied.

Reply:
This is _not_ the same logic as:

```javascript
      const T = b === 1 ? Uint8Array : b === 2 ? Uint16Array : Uint32Array;
      data = rasters.data.map(r => new T(r));
```